nixos/modules/config/swap.nix

```diff
@@ -5,6 +5,52 @@ with lib;
 
 let
 
+  randomEncryptionCoerce = enable: { inherit enable; };
+
+  randomEncryptionOpts = { ... }: {
+
+    options = {
+
+      enable = mkOption {
+        default = false;
+        type = types.bool;
+        description = ''
+          Encrypt swap device with a random key. This way you won't have a persistent swap device.
+
+          WARNING: Don't try to hibernate when you have at least one swap partition with
+          this option enabled! We have no way to set the partition into which hibernation image
+          is saved, so if your image ends up on an encrypted one you would lose it!
+
+          WARNING #2: Do not use /dev/disk/by-uuid/… or /dev/disk/by-label/… as your swap device
+          when using randomEncryption as the UUIDs and labels will get erased on every boot when
+          the partition is encrypted. Best to use /dev/disk/by-partuuid/…
+        '';
+      };
+
+      cipher = mkOption {
+        default = "aes-xts-plain64";
+        example = "serpent-xts-plain64";
+        type = types.str;
```

Comment:
making the cipher enumerable / computable would make this option much more complex because there are lots of different possible values (and those may change over time when cryptsetup gets updates), even though only a few of them are commonly used.

just to give some common examples:
- aes-xts-plain64
- serpent-xts-plain64
- twofish-xts-plain64
- aes-cbc-essiv:sha256
- serpent-cbc-essiv:sha256
- twofish-cbc-essiv:sha256

so basically it is `(aes|serpent|twofish)-(cbc|xts)-(plain64|essiv:[hashalgo])` and depending on cbc or xts the key size varies.

CBC: 128 or 256 bit
XTS: 256 or 512 bit

And depending on compile- and runtime options the available hashing algorithms are SHA1, SHA256, SHA512, ripemd160 and whirlpool